src/NodaTime/Calendars/GJYearMonthDayCalculator.cs

```diff
@@ -15,22 +15,20 @@ internal abstract class GJYearMonthDayCalculator : RegularYearMonthDayCalculator
         protected static readonly int[] MinDaysPerMonth = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
         protected static readonly int[] MaxDaysPerMonth = { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
 
-        private static readonly int[] MinTotalDaysByMonth;
-        private static readonly int[] MaxTotalDaysByMonth;
+        // Note: these fields must be declared after MinDaysPerMonth and MaxDaysPerMonth so that the initialization
+        // is correct. This behavior (textual order for initialization) is guaranteed by the spec. We'd normally
+        // try to avoid relying on it, but that's quite hard here.
+        private static readonly int[] MinTotalDaysByMonth = GenerateTotalDaysByMonth(MinDaysPerMonth);
+        private static readonly int[] MaxTotalDaysByMonth = GenerateTotalDaysByMonth(MaxDaysPerMonth);
 
-        static GJYearMonthDayCalculator()
+        private static int[] GenerateTotalDaysByMonth(int[] monthLengths)
```

Comment:
Add trivial documentation? If I read correctly, this is just Haskell's `scanl1 (+)` with a leading zero, but either a comment here or (maybe better) above the *TotalDaysByMonth arrays would be nice.